### ISSUE TITLE
Different Game.ini/GameUserSettings.ini usage

### ISSUE THREAD

**Issue by waltzie** [FezVrasta/ark-server-tools#989]
Atm i have configfile_xxx to specify where instance config files are but ther's no way to specify to use different Game.ini/GameUserSettings.ini for each instance.
Is there a way to solve this issue?
Some parameter present in Game.ini file are unique and can be different for each instance (e.g. MaxNumberOfPlayersInTribe)

Is there a way to indicate on different instance configuration file to use different Game.ini/GameUserSettings.ini ?

**Comment by klightspeed** [comment]
As far as I know, the game server does not provide any command-line options to specify that a different config file and/or directory be used.

**Comment by waltzie** [comment]
Dear Ben
I'll try to make different users creating links to main steam directories but real local one for Config path

Thanks Walter


Da: "Ben Peddell" notifications@github.com
A: "FezVrasta/ark-server-tools" ark-server-tools@noreply.github.com
Cc: "Walter Summonte" walter@summonte.com,"Author" author@noreply.github.com
Data: Sun, 25 Nov 2018 07:45:20 -0800
Oggetto: Re: [FezVrasta/ark-server-tools] Different Game.ini/GameUserSettings.ini usage (#989)


As far as I know, the game server does not provide any command-line options to specify that a different config file and/or directory be used.

**Comment by waltzie** [comment]
I find the trick ... duplicating the ARK directory and changing the arkserverroot="/home/steam/ARK01" on instance config file!

**Comment by SinisterSole** [comment]
Correct. Each server has to run from it's own directory (unless it's clustered which I have no knowlege in configuring) When you use the cp command in Linux, include the -ar option to preserve attributes and recursive copies.

You can then have multiple config (xxxx.cfg) files to start each server independently as long as they point to the correct folder.

**Comment by TruWrecks** [comment]
Each server needs to be installed in a unique directory, even for clusters. The difference with clusters is that you need to specify a common AltSaveDirectory on all of the servers to point into the same directory and add the map or server name for each unique server.

Ex:
server1: ark_AltSaveDirectoryName="cluster/Aberration"
             arkopt_clusterid="MyCluster12345"

server2: ark_AltSaveDirectoryName="cluster/Scorched"
             arkopt_clusterid="MyCluster12345"

server3: ark_AltSaveDirectoryName="cluster/TheIsland"
             arkopt_clusterid="MyCluster12345"

These entries are the entries needed in the instance files to make a cluster work with arkmanager.

**Comment by CaspyBXL** [comment]
@TrueWrecks 

Can you elaborate something ?
How is an alternative Directory have to do with clusters ?

On many clusters and many servers I have never used an Alternative Save Directory, its not required

You just change the cluster folder for all servers in their instance.cfg and bind their ID 

#clustersettings 
arkopt_ClusterDirOverride="/home/steam/clusters"
arkopt_clusterid=XXXXX
arkflag_NoTransferFromFiltering=true

Is your cluster running fine with just changing "AltSaveDirectoryName" ? 
This is interesting ... makes no sense to me ...

Can you post an example of what you have as folder list in your game ?
I'm curious to see ...
I think you are lucky that official changed its way of making cluster transfers based on unique ID.
Because I tested 9 month's ago with this method ... and got the issue that the serer on launch creates its own cluster folder inside the game folder ... not making it possible for other instances to get to that location ... so how in the world did you manage to get a cluster working on "just" AltSaveDirectoryName
and not even using the ClusterDirOverride

... i'm speechless :-)

**Comment by klightspeed** [comment]
AltSaveDirectoryName is used when the servers share the same install (which requires they have the same settings and set of mods).

Using a single install directory was a hangover from between the introduction of clusters and the introduction of the ClusterDirOverride setting.

If the reason for the single install is due to disk space limitations, arkmanager can use a shared staging directory (on the same filesystem) and hardlinks to reduce the disk space taken.

**Comment by waltzie** [comment]
I've many servers configured with same cluster and different ARK directories (haveing different Mods and Game.ini/GameUserSettings.ini)

The trick is to duplicate the ARK directory (where everithing is stored) 
e.g.
my steam directory:
drwxrwxr-x  6 steam steam 4096 Dec 20 06:11 ARK
drwxrwxr-x  6 steam steam 4096 Dec 12 17:55 ARK2
drwxrwxr-x  6 steam steam 4096 Dec 15 15:43 ARK3
drwxr-xr-x 14 steam steam 4096 Dec 20 00:01 ARK-Backups
-rwxr-xr-x  1 steam steam  104 Dec 10 11:07 ark_daily.sh
-rwxr-xr-x  1 steam steam  732 Dec 19 13:38 ark_hourly.sh
-rwxr-xr-x  1 steam steam   75 Dec 12 08:28 ark_restart.sh
drwxrwxr-x  2 steam steam 4096 Nov 24 00:47 bin
-rwxr-xr-x  1 steam steam  117 Dec 17 10:28 check.sh
drwxrwxr-x  3 steam steam 4096 Nov 24 11:26 clusters
lrwxrwxrwx  1 steam steam   40 Dec 12 12:13 instances -> /home/steam/.config/arkmanager/instances
drwxrwxr-x  3 steam steam 4096 Nov 24 23:21 logs
-rwxr-xr-x  1 steam steam  704 Dec  1 18:37 restart.sh
drwxrwxr-x  9 steam steam 4096 Dec 20 12:00 Steam
drwxr-xr-x  7 steam steam 4096 Dec 20 12:00 steamcmd

as you can see the ARK directory is duplicated and it's instance configuration contain:

#instance ARK ...
...
arkserverroot="/home/steam/ARK" #change to correct path for different mods/ini settings
...
# Cluster Config
arkflag_NoTributeDownloads=False
ark_PreventDownloadSurvivors=False
ark_PreventDownloadItems=False
ark_PreventDownloadDinos=True
ark_PreventUploadSurvivors=False
ark_PreventUploadItems=False
ark_PreventUploadDinos=False
arkflag_NoTransferFromFiltering=""
arkopt_ClusterDirOverride="/home/steam/clusters" #use the same path to share the cluster
arkopt_clusterid="*************"
....
